boot/zephyr/include/sysflash/pm_sysflash.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef __PM_SYSFLASH_H__
+#define __PM_SYSFLASH_H__
+/* Blocking the __SYSFLASH_H__ */
+#define __SYSFLASH_H__
+
+#include <pm_config.h>
+#include <mcuboot_config/mcuboot_config.h>
+
+#ifndef CONFIG_SINGLE_APPLICATION_SLOT
+
+#if (MCUBOOT_IMAGE_NUMBER == 1)
+
+#define FLASH_AREA_IMAGE_PRIMARY(x)    PM_MCUBOOT_PRIMARY_ID
+#define FLASH_AREA_IMAGE_SECONDARY(x)  PM_MCUBOOT_SECONDARY_ID
+
+#elif (MCUBOOT_IMAGE_NUMBER == 2)
+
+/* If B0 is present then two bootloaders are present, and we must use
+ * a single secondary slot for both primary slots.
+ */
+#ifdef PM_B0_ADDRESS
```

Comment:
This doesn't sound right, b0 should be outside the scope of mcuboot entirely. An (actual customer used) design is b0 and mcuboot on app core, b0 -> mcuboot -> user app, then b0 and app on network core where mcuboot on app core can update network core application, I don't see how such an update would work with the below

Reply:
> This doesn't sound right, b0 should be outside the scope of mcuboot entirely. An (actual customer used) design is b0 and mcuboot on app core, b0 -> mcuboot -> user app, then b0 and app on network core where mcuboot on app core can update network core application, I don't see how such an update would work with the below

That is how it was before, I have just moved it outside of sysflash.h.
I think there may be more problems requiring addressing then that, we probably should review what we have, regarding the partition manager and MCUboot, and figure what needs fixing.

Reply:
This is legacy stuff which was added before MCUBoot supported multi-image updates. This is essentially a hack to have mcuboot move a copy of it self into either s0 or s1 depending on where it was linked against. It gets this information through the linker and the symbol _image_1_primary_slot_id. Where this slot id corresponds to the free s0/s1 slot for mcuboot updates.

I have a PR up to fix it for nRF53 but all in all this should be re-designed but it's not a priority currently AFAIK.
Ref.
https://github.com/nrfconnect/sdk-mcuboot/pull/260

Reply:
I don't think we can decouple them completely as b0 depends on MCUBoot doing the flash writing.

Reply:
b0 doesn't depend on anything, it just boots whatever image is valid with the highest version, do you mean b0n for net core?

Reply:
Reverse. MCUBoot depends on having some information about where b0 expects images to be located.

Reply:
This is a real end product static PM file which runs on an nRF5340-based design, first stage bootloader is b0, second stage bootloader is mcuboot, mcuboot supports updating the user application (on the network core) and the application on the network core https://github.com/LairdCP/zephyr_lib/blob/main/mcuboot_config/pm_static.sb.bl5340_dvk_cpuapp.yml how will this work given this file excludes the network core update slot if b0 is present?